sqlmesh/core/engine_adapter/base.py

```diff
@@ -1521,7 +1531,9 @@ def _get_temp_table(self, table: TableName, table_only: bool = False) -> exp.Tab
         Returns the name of the temp table that should be used for the given table name.
         """
         table = t.cast(exp.Table, exp.to_table(table).copy())
-        table.set("this", exp.to_identifier(f"__temp_{table.name}_{random_id(short=True)}"))
+        table.set(
+            "this", exp.to_identifier(f"__temp_{table.name}_{random_id(short=True)}", quoted=True)
```

Comment:
this change doesn't seem to have been necessary because we quote during execution. why did you change this?